rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/interactor/CovidPatientHistoryActivityInteractor.java

```diff
@@ -85,8 +85,17 @@ private String getValues(Obs obs, Map<String, String> formWidgetKeyToTextMap) {
      * @param formWidgetKeyToTextMap
      * @return
      */
-    private String getValue(String value, Map<String, String> formWidgetKeyToTextMap) {
-        return formWidgetKeyToTextMap.containsKey(value) ? formWidgetKeyToTextMap.get(value) : value;
+    private String getValue(String value, Map<String, String> formWidgetKeyToTextMap, Obs obs) {
+
+        if (formWidgetKeyToTextMap.containsKey(value)) {
+            if (StringUtils.isNotBlank(formWidgetKeyToTextMap.get(value))) {
+                return formWidgetKeyToTextMap.get(value);
+            } else {
```

Comment:
Do we need this else? Does it not achieve the same goal as this https://github.com/onaio/rdt-standard/pull/634/files#diff-076ab56e0a4bf66c7d8e493217c0b50ced7501ca39782cf2a6ecadf10b5d1f53R97.

Reply:
yes, we need `obs.getValue()` if we found the empty value.